aca_required_temp.py

```
@@ -0,0 +1,189 @@
+    return opt
+
+
+def get_agasc_cone(ra, dec, time=None, faint_lim=10.8):
+    cone_stars = agasc.get_agasc_cone(ra, dec, date=time)
+    cols = ['AGASC_ID', 'MAG_ACA', 'COLOR1',
+            'RA_PMCORR', 'DEC_PMCORR']
+    ok_cone_stars = cone_stars[(cone_stars['MAG_ACA'] < faint_lim)
+                               & (cone_stars['CLASS'] == 0)
+                               & (cone_stars['ASPQ1'] == 0)
+                               & (cone_stars['COLOR1'] != 0.7)][cols]
+    ok_cone_stars.sort('MAG_ACA')
+    return ok_cone_stars
+
+
+def select_fov_stars(ra, dec, roll, cone_stars):
+    edgepad = EDGE_DIST / 5.
+    q = Quat((ra, dec, roll))
+    yag, zag = radec2yagzag(cone_stars['RA_PMCORR'], cone_stars['DEC_PMCORR'], q)
+    row, col = chandra_aca.yagzag_to_pixels(yag * 3600,
+                                            zag * 3600, allow_bad=True)
+    stars_in_fov = cone_stars[(row < (512 - edgepad))
+                              & (row > (-512 + edgepad))
+                              & (col < (512 - edgepad))
+                              & (col > (-512 + edgepad))]
+    return stars_in_fov
+
+
+def max_temp(ra, dec, roll, time, cone_stars):
+    fov_stars = select_fov_stars(ra, dec, roll, cone_stars)
+    if not len(fov_stars):
+        return None
+    # take the 8 brightest
+    stars = fov_stars[0:8]
+    id_hash = tuple(stars['AGASC_ID'])
+    if id_hash in TEMP_CACHE:
+        t_ccd, n_acq = TEMP_CACHE[id_hash]
+    else:
+        t_ccd, n_acq = t_ccd_warm_limit(date=time,
+                                        mags=stars['MAG_ACA'],
+                                        colors=stars['COLOR1'],
+                                        min_n_acq=N_ACQ_STARS,
+                                        cold_t_ccd=COLD_T_CCD,
+                                        warm_t_ccd=WARM_T_CCD)
+        TEMP_CACHE[id_hash] = (t_ccd, n_acq)
+    return t_ccd
+
+
+def get_rolldev(pitch):
+    if pitch < ROLL_TABLE['pitch'][0]:
```

Should raise an exception since this violates the CARD.  It means the upstream code is doing something wrong.